### PR TITLE
Enable faster sync

### PR DESCRIPTION
**Pull request by antiochp**
RFC proposal for enabling faster sync via "batched" subtrees of the output PMMR.

We have a way forward to verify subtrees of the output PMMR without necessarily having to wait to download the full output and kernel PMMRs. This lets us download subtrees of the output PMMR from multiple peers in parallel, verifying them as they arrive.
Plan is to eventually remove the existing bottleneck around the sync process where we are currently limited to waiting for a single monolithic `txhashset.zip` download from a single, potentially slow or unreliable peer.

Splitting this download across multiple peers and allowing these to happen in parallel will allow the whole sync process to be faster and more robust.

### CODE REVIEW

**Comment by DavidBurkett** [comment]
Rendered: https://github.com/mimblewimble/grin-rfcs/blob/ec0f64ff639792b2a643a13c479bcb4bb70ddc36/text/0000-enable-faster-sync.md

**Comment by antiochp** [comment]
Moved from draft and into review.

**Comment by cadmuspeverell** [comment]
How very ironic.  I offered this exact design many months ago in issue #2387 - and yet it was you who sent me a long cutting message in gitter for a mere mention of adding a UTXO commitment.  I'm elated to see progress finally being made - tho I do wish I wasn't promptly shooed away.  We may have been 6 months closer to seeing these changes in action...

**Comment by antiochp** [comment]
@cadmuspeverell Good to see you around again. I wouldn't call it ironic necessarily - more that we took the time to think it through fully (both #2837 and other similar proposals).

File: text/0000-enable-faster-sync.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+
+- Title: enable-faster-sync
+- Authors: [Antioch Peverell](mailto:apeverell@protonmail.com), [John Tromp](mailto:john.tromp@gmail.com)
+- Start date: Oct 25, 2019
+- RFC PR: Edit if merged: [mimblewimble/grin-rfcs#0000](https://github.com/mimblewimble/grin-rfcs/pull/0000) 
+- Tracking issue: [Edit if merged with link to tracking github issue]
```
Comment (DavidBurkett) [inline text/0000-enable-faster-sync.md]: Please update this to point to: https://github.com/mimblewimble/grin/issues/3173

File: text/0000-enable-faster-sync.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
+
+- Title: enable-faster-sync
+- Authors: [Antioch Peverell](mailto:apeverell@protonmail.com), [John Tromp](mailto:john.tromp@gmail.com)
+- Start date: Oct 25, 2019
+- RFC PR: Edit if merged: [mimblewimble/grin-rfcs#0000](https://github.com/mimblewimble/grin-rfcs/pull/0000)
```
Comment (DavidBurkett) [inline text/0000-enable-faster-sync.md]: This will be RFC 0009

**Pull request merged**